### PR TITLE
les-40-Birthday

### PR DESCRIPTION
**Pull request by Kurilo-Pavel**

### CODE REVIEW

**Review by Terapool — CHANGES_REQUESTED** [review]

**Review by OxygenO22 — APPROVED** [review]

File: src/components/Birthday.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import {useState} from "react";
+import "./birthday.scss";
+
+const Birthday = () => {
+  const [value, setValue] = useState<any>(null);
```
Comment (Terapool) [inline src/components/Birthday.tsx:7]: const [value, setValue] = useState&lt;Date&gt;(null);

File: src/components/Birthday.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const fieldHandler = (event: Date) => {
```
Comment (Terapool) [inline src/components/Birthday.tsx:8]: it is better to name the argument here as date (not event)

const fieldHandler = (date: Date) => {

File: src/components/Birthday.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    setValue(event);
+  }
+
+  return (
+    <div className='birthday'>
+      <label className='birthdaylabel'>When is your birthday?</label>
+      <DatePicker onChange={(e: any) => fieldHandler(e)}
```
Comment (Terapool) [inline src/components/Birthday.tsx:15]: onChange={(e: Date) => fieldHandler(e)}

File: src/components/Birthday.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import {useState} from "react";
+import "./birthday.scss";
+
+const Birthday = () => {
```
Comment (Terapool) [inline src/components/Birthday.tsx:6]: The name of the component is not clear by its meaning. 

I would recommend to name it "BirthdayPicker", "BirthdayInput", "DayPicker", "DaySelector", etc.

File: src/components/birthday.scss [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  display: block;
+  font-weight: bold;
+  margin: 5px 0;
+}
+
+.birthdayInput {
+  width: 100%;
+  box-sizing: border-box;
+  padding: 0.5em;
+  border: 1px solid blue;
```
Comment (Terapool) [inline src/components/birthday.scss:17]: Do you use here the standard css blue color or your variable?
If variable you must write
 border: 1px solid $blue;
 
 If you use standard blue color so you do not need variable at line 1...

File: src/components/birthday.scss [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+.error {
+  color: red;
+  display: block;
+  margin-top: 0.5em;
+}
+
+.react-datepicker__current-month{
+  visibility: hidden;
+}
```
Comment (Terapool) [inline src/components/birthday.scss:34]: Please make consistency: an empty line before each class declaration

**Review by Terapool — CHANGES_REQUESTED** [review]

File: src/components/Birthday.tsx (hunk 3 above)
Comment (Kurilo-Pavel) [reply src/components/Birthday.tsx:15]: first value is "null"

Comment (Terapool) [reply src/components/Birthday.tsx]: See the comment above

**Review by DimitarShemet — APPROVED** [review]

File: src/components/BirthdayPicker.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+  return (
+    <div className='birthday'>
+      <label className='birthdaylabel'>When is your birthday?</label>
+      <DatePicker onChange={(e: Date) => fieldHandler(e)}
+                  selected={value}
+                  placeholderText='--.--.----'
+                  dateFormat='dd.MM.yyyy'
+                  maxDate={new Date()}
+                  minDate={new Date(minYear)}
```
Comment (Terapool) [inline src/components/BirthdayPicker.tsx:21]: no need to make new Date from minYear, simply minYear:

minDate={minYear}

File: src/components/BirthdayPicker.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import "react-datepicker/dist/react-datepicker.css";
+import {useState} from "react";
+import "./birthday.scss";
+
+const BirthdayPicker = () => {
+  const [value, setValue] = useState<Date | null>(null);
+  const fieldHandler = (date: Date) => {
+    setValue(date);
+  }
+  const minYear = new Date((new Date()).getTime() - 120 * 365.2 * 86400000 - 6 * 86400000 + 29.7 * 60000);
```
Comment (Terapool) [inline src/components/BirthdayPicker.tsx:11]: too complex calculations to understand what minYear is expected to be...

What is expected to be here?

Comment (Kurilo-Pavel) [reply src/components/BirthdayPicker.tsx:11]: I need Date 120 years ago, every day date should chenge

Comment (Terapool) [reply src/components/BirthdayPicker.tsx:11]: For such puprorses it is recommended to use 'momentjs' library.

Please, leave commented description that it is 120 years ago date

**Review by Terapool — APPROVED** [review]

File: src/components/BirthdayPicker.tsx (hunk 8 above)
Comment (Kurilo-Pavel) [reply src/components/BirthdayPicker.tsx]: Ok, Thanks)

File: src/components/BirthdayPicker.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+                  selected={value}
+                  placeholderText='--.--.----'
+                  dateFormat='dd.MM.yyyy'
+                  maxDate={new Date()}
+                  minDate={minYear}
+                  showMonthDropdown
+                  showYearDropdown
+                  dropdownMode="select"
+                  className='birthdayInput'
+                  required/>
```
Comment (Evgeniy121212) [inline src/components/BirthdayPicker.tsx:27]: Please, be sure, that label "linked" to the DataPicker

There is 2 ways to do that:
- `<label><DatePicket/></label>`
- `<label></label> <DatePicket/>`, but with settled `for`-property for the label

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

**Pull request merged**